crates/nu-parser/src/parser.rs

```diff
@@ -1030,6 +1038,80 @@ impl<'a> ParserWorkingSet<'a> {
         }
     }
 
+    pub fn parse_range(&mut self, token: &str, span: Span) -> (Expression, Option<ParseError>) {
+        let dotdot_pos: Vec<_> = token.match_indices("..").map(|(pos, _)| pos).collect();
+
+        if dotdot_pos.len() != 1 {
+            return (
+                garbage(span),
+                Some(ParseError::Expected("range".into(), span)),
+            );
+        }
+
+        let (bounds, operator_str, operator): (Vec<_>, _, _) = if let Some(pos) = token.find("..<")
+        {
+            if pos == dotdot_pos[0] {
+                (
+                    token.split("..<").collect(),
+                    "..<",
+                    RangeOperator::RightExclusive(Span::new(
+                        dotdot_pos[0],
+                        dotdot_pos[0] + "..<".len(),
+                    )),
+                )
+            } else {
+                return (
+                    garbage(span),
+                    Some(ParseError::Expected("range".into(), span)),
+                );
+            }
+        } else {
+            (
+                token.split("..").collect(),
+                "..",
+                RangeOperator::Inclusive(Span::new(dotdot_pos[0], dotdot_pos[0] + "..".len())),
+            )
+        };
+
+        if bounds.len() != 2 {
+            return (
+                garbage(span),
+                Some(ParseError::Expected("range".into(), span)),
+            );
+        }
+
+        let span_lhs = Span::new(span.start, span.start + dotdot_pos[0]);
+        let lhs = match self.parse_number(bounds[0], span_lhs) {
```

Comment:
I think for this you want to recursively parse the expressions on either side. For example, it could be `(2 + 3)..(5 + 9)` and both sides would have a subexpression